server/repository/repository.go

```diff
@@ -198,7 +197,14 @@ func (s *Server) ListRepositories(ctx context.Context, q *repositorypkg.RepoQuer
 				Project:            repo.Project,
 				ForceHttpBasicAuth: repo.ForceHttpBasicAuth,
 				InheritedCreds:     repo.InheritedCreds,
-			})
+			}
+
+			if q.Repo != "" && q.Repo == repo.Repo {
+				items = append(items, repository)
+				continue
+			}
+			// remove secrets
+			items = append(items, repository)
```

Comment:
I did not really understand the difference this code is bringing. Could you give an example?